server/k8s.go

```diff
@@ -154,6 +212,45 @@ func buildDetails(service *v1.Service, externalServiceName string) *routableServ
 	rs := &routableService{
 		externalServiceName: externalServiceName,
 		containerEndpoint:   net.JoinHostPort(clusterIp, port),
+		autoScaleUp:         w.buildScaleUpFunction(service),
 	}
 	return rs
 }
+
+func (w *k8sWatcherImpl) buildScaleUpFunction(service *core.Service) func(ctx context.Context) error {
+	return func(ctx context.Context) error {
+		serviceName := service.Name
+		if statefulSetName, exists := w.mappings[serviceName]; exists {
+			if scale, err := w.clientset.AppsV1().StatefulSets(service.Namespace).GetScale(ctx, statefulSetName, meta.GetOptions{}); err == nil {
+				replicas := scale.Status.Replicas
+				logrus.WithFields(logrus.Fields{
+					"service":     serviceName,
+					"statefulSet": statefulSetName,
+					"replicas":    replicas,
+				}).Debug("StatefulSet of Service Replicas")
+				if replicas == 0 {
+					if _, err := w.clientset.AppsV1().StatefulSets(service.Namespace).UpdateScale(ctx, statefulSetName, &autoscaling.Scale{
```

Comment:
I'm thinking the reason [this is happening](https://github.com/itzg/mc-router/issues/21#issuecomment-974713622) is because updating the scale is an asynchronous operation and the backend connection is attempted immediately in `findAndConnectBackend`. 

Over in `findAndConnectBackend`, it probably needs to have a retry loop with a delay between each attempt to connect to the backend. It's possibly overkill, but I have been pleased with [this library](https://pkg.go.dev/github.com/cenkalti/backoff/v4) for robust handling of retry logic.

Reply:
@itzg would you be open to leave [resolving this](https://github.com/itzg/mc-router/issues/21#issuecomment-974713622) to a future follow-up PR instead of this one, and consider accepting this once the other feedback here will have been addressed? -- As for the retry logic, I'm thinking that an [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff) will delay connecting the user unnecessarily... that kind of approach makes sense to avoid network congestion and backend overload and cascading failures - but that's not our scenario here, do you agree? We just want the (one and only, ever, single) `mc-router` "client" to the connect to that Minecraft server ASAP - which a backoff just delays. Would a basic loop trying for N times with M wait time, perhaps with M=100ms and N globally configurable on the CLI perhaps be both simpler and better?

Reply:
Sure that aspect can be delayed. My suggestion of a retry mechanism didn't mean to imply the use of exponential backoff. Fixed retry interval, as you mentioned, was what I expected.